### ISSUE TITLE
Docker Container Crashes When Out of Memory

### ISSUE THREAD

**Issue by dyno10** [qbittorrent/docker-qbittorrent-nox#34]
### qBittorrent & operating system versions

qBittorrent: 4.6.4 x64
Operating system: Unraid 6.12.6
Qt: 6.6.1
libtorrent-rasterbar: 1.2.19.0
Docker Container:  Official qBittorrent Docker

### What is the problem?

This Docker container crashes when qBittorrent-nox uses all available memory.  Instead, the container should invoke OOM Killer to reap the qBittorrent-nox process and restart the process.  This is how other qBittorrent Docker containers (from other maintaners) handle this issue.  

This happens because I set a 64GiB memory limit for this Docker container, to prevent it from using all system RAM (and crashing my server).  

![docker](https://github.com/qbittorrent/docker-qbittorrent-nox/assets/166157675/c391c219-63e4-4130-a7fe-8f82b96348f0)

This is the first time I've had a container crash, but perhaps that's just a difference in the way different Docker containers handle OOM events.  The container log says:

----

terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc

*************************************************************
Please file a bug report at https://bug.qbittorrent.org and provide the following information:

qBittorrent version: v4.6.4

Caught signal: SIGABRT
```
 0# getStacktrace[abi:cxx11]() in /usr/bin/qbittorrent-nox
 1# 0x000055C472C1AEDA in /usr/bin/qbittorrent-nox
 2# 0x000014B410F87EA8 in /lib/ld-musl-x86_64.so.1
```

Related Issue:  https://github.com/qbittorrent/qBittorrent/issues/20675

**Comment by Chocobo1** [comment]
> Instead, the container should invoke OOM Killer to reap the qBittorrent-nox process and restart the process.

A qbt crash may be the result of other issues (for example, security issues) and IMO it is better to have it manually inspected before a restart.

**Comment by Chocobo1** [comment]
If anyone requires it, use [`--restart`](https://docs.docker.com/config/containers/start-containers-automatically/) flag.

**Comment by Arrmm** [comment]
I am also having a similar issue except in my case I have only allocated 4GB RAM instead of 64.  Is there a minimum limit?  I have over 300 torrents seeding.

The Docker image would run fine for anything between 30 to 90 mins and it crashes.  I tried to look into the docker logs but there is nothing shown.  Is there anywhere else I can look into to see if there might be settings that I need to optimize?
